### ISSUE TITLE
Is there any way to solve this"Encountered internal error running command"

### ISSUE THREAD

**Issue by liubo-ricoh** [appium/appium#17826]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the appium is running, the "Encountered internal error running command" appears in the log

### Expected Behavior

Hopes this internal problem will not occur

### Minimal Reproducible Example

2022-11-04 10:36:19:228 [WD Proxy] Got response with status 200: {"sessionId":"89f0b658-08ad-4fb7-8bbd-590571e94e2f","value":[]}
2022-11-04 10:36:19:228 [W3C (492d0506)] Responding to client with driver.findElements() result: []
2022-11-04 10:36:19:229 [HTTP] <-- POST /wd/hub/session/492d0506-be18-4df6-b458-cc712ce6037d/elements 200 10550 ms - 12
2022-11-04 10:36:19:229 [HTTP]
2022-11-04 10:36:19:241 [HTTP] --> POST /wd/hub/session/492d0506-be18-4df6-b458-cc712ce6037d/element/7ce2531d-eb35-4e9c-85b1-b9752bfbaf59/click
2022-11-04 10:36:19:241 [HTTP] {"id":"7ce2531d-eb35-4e9c-85b1-b9752bfbaf59"}
2022-11-04 10:36:19:242 [W3C (492d0506)] Calling AppiumDriver.click() with args: ["7ce2531d-eb35-4e9c-85b1-b9752bfbaf59","492d0506-be18-4df6-b458-cc712ce6037d"]
2022-11-04 10:36:19:243 [WD Proxy] Matched '/element/7ce2531d-eb35-4e9c-85b1-b9752bfbaf59/click' to command name 'click'
2022-11-04 10:36:19:243 [WD Proxy] Proxying [POST /element/7ce2531d-eb35-4e9c-85b1-b9752bfbaf59/click] to [POST http://127.0.0.1:8279/wd/hub/session/89f0b658-08ad-4fb7-8bbd-590571e94e2f/element/7ce2531d-eb35-4e9c-85b1-b9752bfbaf59/click] with body: {"element":"7ce2531d-eb35-4e9c-85b1-b9752bfbaf59"}
2022-11-04 10:36:19:741 [WD Proxy] Got response with status 404: {"sessionId":"89f0b658-08ad-4fb7-8bbd-590571e94e2f","value":{"error":"stale element reference","message":"Cached elements 'By.xpath: //*[@text='XXXX screen in']' do not exist in DOM anymore","stacktrace":"io.appium.uiautomator2.common.exceptions.StaleElementReferenceException: Cached elements 'By.xpath: //*[@text='XXXX screen in']' do not exist in DOM anymore\n\tat io.appium.uiautomator2.model.ElementsCache.restore(ElementsCache.java:72)\n\tat io.appium.uiautomator2.model.ElementsCache.get(ElementsCache.java:145)\n\tat io.appium.uiautomator2.handler.Click.safeHandle(Click.java:36)\n\tat io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:41)\n\tat io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:261)\n\tat io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:255)\n\tat io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:68)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChan...
2022-11-04 10:36:19:741 [W3C] Matched W3C error code 'stale element reference' to StaleElementReferenceError
2022-11-04 10:36:19:742 [W3C (492d0506)] Encountered internal error running command: io.appium.uiautomator2.common.exceptions.StaleElementReferenceException: Cached elements 'By.xpath: //*[@text='XXXX screen in']' do not exist in DOM anymore
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.model.ElementsCache.restore(ElementsCache.java:72)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.model.ElementsCache.get(ElementsCache.java:145)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.handler.Click.safeHandle(Click.java:36)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:41)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:261)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:255)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:68)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
2022-11-04 10:36:19:742 [W3C (492d0506)]     at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:611)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:552)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:466)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:438)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
2022-11-04 10:36:19:743 [W3C (492d0506)]     at java.lang.Thread.run(Thread.java:919)
2022-11-04 10:36:19:743 [W3C (492d0506)]
2022-11-04 10:36:19:743 [HTTP] <-- POST /wd/hub/session/492d0506-be18-4df6-b458-cc712ce6037d/element/7ce2531d-eb35-4e9c-85b1-b9752bfbaf59/click 404 502 ms - 3967
2022-11-04 10:36:19:743 [HTTP]

### Environment

- Operating system:
- If running via `appium` CLI...
  - Appium CLI version (output of `appium --version`):
  - Node.js version (output of `node --version`):
  - `npm` version (output of `npm --version`): 
  - Last `appium` version which did _not_ exhibit the problem: 
- If running Appium Desktop, its version:
- Appium driver(s) and their version(s):
- Appium plugin(s) and their version(s):
- Platform and version under test:
- Real device or emulator/simulator:

**Comment by mykola-mokhnach** [comment]
> Cached elements 'By.xpath: //*[https://github.com/text='XXXX screen in']' do not exist in DOM anymore

What exactly is unclear about this particular error?

Possible duplicate of https://github.com/appium/appium/issues/15348

**Comment by liubo-ricoh** [comment]
It does not mean 100% and may be successful. Only once in three times

**Comment by mykola-mokhnach** [comment]
I mean this exception is the expected driver behavior in case the cached element is not present in the view at the moment it is accessed. I don't see any issues there.

**Comment by mykola-mokhnach** [comment]
Closed because of no response
